librosa/core/pitch.py

```diff
@@ -389,31 +384,25 @@ def _cumulative_mean_normalized_difference(
     yin_frames : np.ndarray [shape=(max_period-min_period+1,n_frames)]
         Cumulative mean normalized difference function for each frame.
     """
-    # Autocorrelation.
-    a = np.fft.rfft(y_frames, frame_length, axis=-2)
-    b = np.fft.rfft(y_frames[..., win_length:0:-1, :], frame_length, axis=-2)
-    acf_frames = np.fft.irfft(a * b, frame_length, axis=-2)[..., win_length:, :]
-    acf_frames[np.abs(acf_frames) < 1e-6] = 0
+    acf_frames = audio.autocorrelate(y_frames, max_size=max_period + 1, axis=-2)
 
     # Energy terms.
-    energy_frames = np.cumsum(y_frames**2, axis=-2)
-    energy_frames = (
-        energy_frames[..., win_length:, :] - energy_frames[..., :-win_length, :]
-    )
-    energy_frames[np.abs(energy_frames) < 1e-6] = 0
+    yin_frames = np.cumsum(np.square(y_frames), axis=-2)
```

Comment:
not for this PR, but cumsum(square) composition seems like a good candidate for numba ufunc optimization.

Reply:
Actually, we can cut memory consumption in half here with an in-place cumsum.  No further optimization needed.
```suggestion
    yin_frames = np.square(y_frames)
    np.cumsum(yin_frames, out=yin_frames, axis=-2)
```

Reply:
Thanks! I'll add in this change.